hudi-common/src/main/java/org/apache/hudi/common/model/OverwriteWithLatestAvroPayload.java

```diff
@@ -50,10 +50,6 @@ public OverwriteWithLatestAvroPayload(Option<GenericRecord> record) {
 
   @Override
   public OverwriteWithLatestAvroPayload preCombine(OverwriteWithLatestAvroPayload oldValue) {
-    if (oldValue.recordBytes.length == 0) {
-      // use natural order for delete record
-      return this;
```

Comment:
Shouldn't this be guarded by changelog config?

Reply:
Also, I think a better way would be to subclass this and create a separate implementation for CDC payload. What do you think?
cc @danny0405

Reply:
I agree w/ sagar. 
@danny0405 : what do you think.